packages/react-components/src/hooks/useAlarms/utils/queryUtils.ts

```diff
@@ -1,6 +1,28 @@
 import { UseQueryResult } from '@tanstack/react-query';
 import { AlarmDataStatus } from '../types';
 
+/**
+ * Combine two query statuses
+ *
+ * If any one status isLoading/Refetching/Error is true then the running status is also true
+ * If any one status isSuccess is false then running status is also false
+ * Combine all errors into a running list
+ */
+// Combine two query statuses
+const combineStatuses = ({
+  oldStatus,
+  newStatus,
+}: {
+  oldStatus: AlarmDataStatus;
+  newStatus: AlarmDataStatus;
+}): AlarmDataStatus => ({
+  isLoading: oldStatus.isLoading || newStatus.isLoading,
+  isRefetching: oldStatus.isRefetching || newStatus.isRefetching,
+  isSuccess: oldStatus.isSuccess && newStatus.isSuccess,
+  isError: oldStatus.isError || newStatus.isError,
+  errors: [...(oldStatus.errors ?? []), ...(newStatus.errors ?? [])],
```

Comment:
What do you think about having a single error field too which is a concatenation of all the errors into a single new error? The resource explorer variants take a single error object, so I ended up doing that concatenation. But I'm wondering if other consumers like the charts will also prefer that convenience

Reply:
We should discuss what is the best practice for handling multiple errors. Do we provide like a stack trace of every error in one message? Do we just dump a list of errors as a single object?